src/Operation/Unpack.php

```diff
@@ -21,22 +21,19 @@
 /**
  * @immutable
  *
- * @template NewTKey
- * @template NewT
- *
  * @template TKey
- * @template T of array{0: NewTKey, 1: NewT}
+ * @template T
  */
 final class Unpack extends AbstractOperation
 {
     /**
      * @pure
      *
-     * @return Closure(Iterator<TKey, T>): Generator<NewTKey, NewT>
+     * @return Closure(Iterator<mixed, mixed>): Generator<TKey, T>
```

Comment:
this change is not beneficial because it's restricting the types of the keys and values to the original TKEY, T, which is not the case typically with `unpack` in practice.

I guess if we want to get rid of the other template parameters (because they cannot be inferred), we will have to go with `mixed` here, e.g. `Generator<mixed, mixed>`.

Also, any changes made here should also be reflected in the interface Unpackable and in the static analysis checks

Reply:
Going to do it in here this evening then ! Thanks for the feedback :)🙏